### ISSUE TITLE
Theme Idea (Enhancement)

### ISSUE THREAD

**Issue by Bboy486** [MattsShack/Plex-Movie-Poster-Display#27]
An idea for a theme would be to mimic what Steam or playnite (gaming front-ends) do where they have the graphic in the background and have the poster be the focal point. It could pull from the media folder that plex creates for the background image. This would let this script to be used in different uses that just a black background.

ex. Radarr - https://prnt.sc/ve4cip

**Comment by ZeroCoolGOS** [comment]
We have started on the backend code for this.

Hopefully we will have a front end option in the coming weeks.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

This feature should now be integrated and you should be able to test it out.

**Comment by Bboy486** [comment]
Should I rerun git clone or just apt update?

**Comment by ZeroCoolGOS** [comment]
git pull to get the latest code if your already in the dev branch.

**Comment by Bboy486** [comment]
> git pull to get the latest code if your already in the dev branch.

Sorry I did that but I still see this in the admin

Version 2.7.1  Download Updates

**Comment by ZeroCoolGOS** [comment]
After you do your git clone you still have to copy the files from the git folder to your nginx folder (var/www/html).

We still don't have a rolling/simple upgrade process on active sites just yet.

**Comment by Bboy486** [comment]
Yes I do. I backup my config before I did that. 

pi@raspberrypi:~/Plex-Movie-Poster-Display $ sudo git pull
Already up to date.

sudo git checkout dev
Already on 'dev'
Your branch is up to date with 'origin/dev'.


Maybe let me know what I should be seeing?

**Comment by ZeroCoolGOS** [comment]
If you load the page and you see v2.8.0 or higher then you should have the latest version.

And if your on that version, you can go to the "Coming Soon" and "Now Showing" and show advanced settings and then you should have a background option.

**Comment by Bboy486** [comment]
Ok. I had to run it a few times but it works. 

It is really cool. Looks like it doesn't always take the same photo as the movie (one time it did though).

Is it possible to have the background blurred out or have a cor overlay with screen or multiply on it? 

Ex. https://images.app.goo.gl/nBtcy6ybfPv46zFNA

https://images.app.goo.gl/zccomhujZ5Ta7FVH7

https://images.app.goo.gl/vjXhF13Q5K8rNFqeA


One error when I updated the coming sound text (I change top fro. Title to Coming soon and bottom to title) even though I had the background checked it no longer displays.

**Comment by ZeroCoolGOS** [comment]
When you say the same photo are you expecting the poster also be the background?

I will have too look into how it can blur out the background.

Please elaborate when you say core overlay with screen or multiple on it.

Refresh your page and see if the background switch is still on.

Sometimes when you have made a change, saved then make another change and save, some of the options are overwritten.  This is something I am still working on trying to address.  In the mean time I would just refresh the page before you make an additional change.  This will reload the settings again.

**Comment by ZeroCoolGOS** [comment]
I will add the blur in my next code release for @MattsShack  to review sometime this week.

**Comment by Bboy486** [comment]
When you say the same photo are you expecting the poster also be the background?

No I liked that it was a different but some of the posters had the same image and background image and others had different images. I prefer the later.

**Comment by Bboy486** [comment]
Please elaborate when you say core overlay with screen or multiple on it.

This is a photoshop terms

https://www.photoshopessentials.com/photo-editing/layer-blend-modes/intro/

**Comment by Bboy486** [comment]
Sometimes when you have made a change, saved then make another change and save, some of the options are overwritten. This is something I am still working on trying to address. In the mean time I would just refresh the page before you make an additional change. This will reload the settings again

I did save and refresh but it seems like a big because even though the background image was still checked when I update the settings the front end no longer showed the image. I had to turn it off and back on again so it would take.

**Comment by ZeroCoolGOS** [comment]
> When you say the same photo are you expecting the poster also be the background?
> 
> No I liked that it was a different but some of the posters had the same image and background image and others had different images. I prefer the later.

The way Plex provides the information, we do not have control as to what the background will be for be able to randomize that background.  

If Plex has a background that is different then the poster then it will be used; if Plex's background is that poster then it will be used.  It all has to do with the data from Plex.  If no background is defined for the media, then Plex does not provide data for it.

I hope that made sense.  If not please let me know and I will work to clarify (and update the documentation)

**Comment by ZeroCoolGOS** [comment]
> Please elaborate when you say core overlay with screen or multiple on it.
> 
> This is a photoshop terms
> 
> https://www.photoshopessentials.com/photo-editing/layer-blend-modes/intro/

I have made some adjustments for the background and will be available in the next targeted v2.9 release that I am working on and hope to have completed by the end of the week.  It will have an 8px blur on the background image as it currently stands.

**Comment by ZeroCoolGOS** [comment]
> Sometimes when you have made a change, saved then make another change and save, some of the options are overwritten. This is something I am still working on trying to address. In the mean time I would just refresh the page before you make an additional change. This will reload the settings again
> 
> I did save and refresh but it seems like a big because even though the background image was still checked when I update the settings the front end no longer showed the image. I had to turn it off and back on again so it would take.

I still have yet to fix the bug about the refresh.  But hopefully with the above mentioned v2.9 release the background image will be a bit more stable as I have made some changes to the implementation.

Also sometimes the media may not have background assigned to it in Plex, so in that place a black background will show up.

You should never see another media's background with a different poster.  If that is the case please log a new issue for us to investigate.

**Comment by Bboy486** [comment]
_You should never see another media's background with a different poster. If that is the case please log a new issue for us to investigate._

To be clear I'm not seeing anothers media I am seeing the same background and the poster. Most of the time it has different backgrounds but a few times it shows the same image in both areas.

**Comment by ZeroCoolGOS** [comment]
Hmm. I don't know how that would happen or how we can address it from our end.

I have however seen that sometimes the wrong background art for the poster while using caching. That will be addressed with the next release.

Can you validate on your plex server to see if the poster image is the same as the background?

Also do you have caching on?

**Comment by ZeroCoolGOS** [comment]
The latest dev branch now has the background focus request.

**Comment by Bboy486** [comment]
> Hmm. I don't know how that would happen or how we can address it from our end.
> 
> I have however seen that sometimes the wrong background art for the poster while using caching. That will be addressed with the next release.
> 
> Can you validate on your plex server to see if the poster image is the same as the background?
> 
> Also do you have caching on?

I do have cache on. https://prnt.sc/101khmp****

Where would I see the background image?


Good - https://prnt.sc/101ki2w
Not good - https://prnt.sc/101kjdd

**Comment by ZeroCoolGOS** [comment]
When you go into plex you will have the option on the media when you try and edit it.

http://prnt.sc/101koxp

If you go into your media you will see what the background image is set to and if it is the same as the poster.  You can also change the image if you like.

**Comment by Bboy486** [comment]
I'll look into automating that. Here is another issue. The way the images render are progressive and if I set a low number for transition it doesn't render correctly 

https://drive.google.com/file/d/1XZB0S9iIwXNvFstKCP7Ei5bGvRy1FUR5/view?usp=drivesdk

**Comment by ZeroCoolGOS** [comment]
What part will you automate?  Please keep us posted on any automation you do on your plex because we are always interested in seeing what others are working on :)

As for the progressive image rendering.  Based on your video it looks like you are not using the latest code (as the background does not have a blur).  Could you update to the latest version and see if that fixes the issue?

When we implemented the blur we made some changes in how it would render so that may resolve the issue.  If not let us know and we can try and look into it further.

One other possible issue could be that the images are relatively large for the device to render fast enough and scale down.

But I would recommend upgrading to the latest build first.

**Comment by Bboy486** [comment]
_What part will you automate? Please keep us posted on any automation you do on your plex because we are always interested in seeing what others are working on :)_

I updated my Plex library to the newer agent to see if it does a better job of matching the metadata and images. https://forums.plex.tv/t/introducing-the-new-plex-movie-agent/615989

_As for the progressive image rendering. Based on your video it looks like you are not using the latest code (as the background does not have a blur). Could you update to the latest version and see if that fixes the issue?_

Ah, I keep forgetting to copy to the HTML folder. Adding this for my future reference:
cd Plex-Movie-Poster-Display

cd /var
cd www
cd /html/
sudo nano config.php

Save this file as a backup

cd
cd Plex-Movie-Poster-Display
sudo git checkout dev
sudo get pull
sudo cp -R * /var/www/html

cd /var
cd www
cd html

sudo nano configbackup.php
save as
sudo nano config.php

_One other possible issue could be that the images are relatively large for the device to render fast enough and scale down._

I tried this on both the fire tablet in the video and my laptop and it was the same issue.

**Comment by ZeroCoolGOS** [comment]
To save you time for the backup/restore we have a backup option built into the admin page.

Go to the general page and show the advanced options and there is an "Export Configuration" to backup the file, and a "Restore Configuration" so all you have to do is upload your config file back to the system.

That should help speed up the rollout.

We still need to come up with a better upgrade path/script to make it faster for most people.  For me, I have it in a docker image so it takes no time for me to spin up the latest update.

**Comment by Bboy486** [comment]
I saw that afterwards. Unfortunately I after copying over to the html folder I am greeted with only a back screen.

**Comment by ZeroCoolGOS** [comment]
After you copied all the files your given a blank screen?  That is after your overwrite of the config.php?

Are you able to get into the config pages?

**Comment by Bboy486** [comment]
Yes I can get to the config but nothing on the front end

**Comment by Bboy486** [comment]
Anything else I should try or just delete and reinstall?

**Comment by ZeroCoolGOS** [comment]
I'm not sure what could be the issue. 

The fastest might be to delete and reinstall.

Sorry. I am currently working on this blocking issue we are having. Once that has been fully resolved I will try and get back on creating a better install/update plan to help simplify.

**Comment by Bboy486** [comment]
Sure thing.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

I have now seen the issue your seeing about the blank page.

I will try and get you a hotfix tomorrow.  I know of "what" the issue is, but don't know why just yet.

Sorry for the trouble.

@MattsShack 
I will work on a hot fix and then switch back to the rebranding as I am almost done.

**Comment by Bboy486** [comment]
I went and deleted and reinstalled. New issue is the background image no longer works. Toggling on and off on mobile and saving doesn't update. 

Same on desktop.

Also ip/admin still goes to the old admin. Ip/settings/coming soon.php goes to the new one.

**Comment by ZeroCoolGOS** [comment]
Ok.  Still working on a fix. 

I don't know what happened but something must have gone out of alignment with v2.9. 

Even my logging system broke.  I will try and get it fixed today.

The ip/admin page should not show up but I will try and see what is going on in that one too as part of this fix.

Sorry again, I don't know how/why this got past any of my testing.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 
I have submitted a fix to this issue, now just waiting on @MattsShack to approve.

Once done you need to be running v2.9.3 for this fix and the ip/admin.php should not be there if you have done a clean install.

Once you have this version please let me know if you are still having issues.

**Comment by Bboy486** [comment]
FYI the background art was also missing.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 
Was or still is with v2.9.3?

**Comment by Bboy486** [comment]
Yes. I am on the latest dev 
pi@raspberrypi:~/Plex-Movie-Poster-Display $ sudo git pull
Already up to date.

**Comment by ZeroCoolGOS** [comment]
Ok, sorry.  I will look into this shortly.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

I might have partially replicated your issue with the latest dev branch.  But I found it to be the cache has not been refreshed once the code was updated.  So a Ctrl+F5 should help.

Give that a shot and let me know if that helps.

And just double checking that even though you did the git pull you also copied the files up to the var/www/html path?

**Comment by Bboy486** [comment]
Yes. I was just letting you know I'm on the latest update. I had copied everything over and that's when these issue developed.

**Comment by ZeroCoolGOS** [comment]
Hmm ok. 

I will dig deeper on my end and see if I can some how reproduce.

**Comment by Bboy486** [comment]
Will this help https://www.awesomescreenshot.com/video/2978389?key=98c93be7bda111458e28d2cc1485269e

**Comment by ZeroCoolGOS** [comment]
I know there is a refresh bug when you press the save button. 

So once you hit save you have to hit F5 or go and come back to that page to see the settings update. 

The save button works it's just how it's loading the change after save.

So if you keep making a change on the page, save and make more changes without refresh then the previous changes get over written.

I have yet to figure out this bug and squash it. 

Sorry for the inconvenience.

**Comment by Bboy486** [comment]
Cool. Well ill just wait for the next push.

**Comment by ZeroCoolGOS** [comment]
I don't know if I will be able to fix the refresh bug by the next push.

But in the mean time can you check your config.php file and see if "$comingSoonBackgroundArt = '1';" after you have turned on the background art and saved.

If the config says 1 then the background art should be working.  If it still says 1 and there is no background art then please let me know so I will need to dig more into it and add some more logging to diagnose the problem deeper.

**Comment by Bboy486** [comment]
//Coming Soon Configuration
  $comingSoonBackgroundArt = '1'; //Default: false

https://prnt.sc/10fnlz6

**Comment by ZeroCoolGOS** [comment]
Hmm.  That is odd.
I will keep looking to see what could be the issue and hopefully I will solve the issue for you.

**Comment by titelies** [comment]
FYI, Background Art is working for me using 2.9.3 when testing on my laptop. I don't use Background Art myself cuz my TV is hung vertically and that's where I use PMPD. Just trying to help. https://i.imgur.com/ehLv7Jo.png

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

We have made a few more releases. 

Once you have a chance to update, please let us know if your background is working correctly now. 

Thank you

**Comment by Bboy486** [comment]
I am updated but I am just getting a black screen. 

Steps
cd
cd Plex-Movie-Poster-Display
sudo git checkout dev
sudo git pull
sudo cp -R * /var/www/html

cd /var
cd www
cd html

restore config.php from local folder

sudo chmod -R 774 /var/www/html/
sudo chown -R pi:www-data /var/www/html/


I should note this is the same pi that is running Pihole/PiVPN. Pihole uses the html folder as well but hasn't been an issue before. 

I just cleared cache images and toggled it off and on.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

Yes. another user has also reported an issue with a black screen.   I am working on a fix and hope to have it ready this afternoon.

If you don't want to wait. It seems that if you reenter your config information clean then it will be fine.

**Comment by titelies** [comment]
@Bboy486 I got the black screen as well when updating to the latest 2.10.5 and restoring my config. I had to redo the 2.10.5 update but then I manually put all of the config entries and now I'm working successfully with 2.10.5. I believe it's something with importing the saved config file. 2.10.4 works for me with importing the config file.

**Comment by ZeroCoolGOS** [comment]
I believe it has to do with the refresh values that went into 2.10.5. I have a fix in the works and am trying to test/duplicate with an older config.

**Comment by Bboy486** [comment]
> If you don't want to wait. It seems that if you reenter your config information clean then it will be fine.

I tried this and it did not work. 

Also checkboxes for caches images and any changes need to be checked, saved, checked again as it is not registering on save.

**Comment by ZeroCoolGOS** [comment]
> > If you don't want to wait. It seems that if you reenter your config information clean then it will be fine.
> 
> I tried this and it did not work.
> 
> Also checkboxes for caches images and any changes need to be checked, saved, checked again as it is not registering on save.

The changes are taking the first time.  There is a bug when the page reloads after a save, that the change is not reflected right away.  A second reload of the page will yield the correct values.

I still have not found a fix for this issue.  I believe it might have something to do with the growing number of options we have, and we may need to look at moving to a new file type for the config values.

I am hoping that the v.2.10.6 will resolve the restore issue, as I have not been able to recreate the issue you guys are having but I thinking I had a good idea as to where the bug was.

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

As if 2.11.x I have adjusted the time issue with saving and seeing the changes so now you should be able to save and when it reloads, your changes should be reflected.

**Comment by Bboy486** [comment]
The save seems to work now. I still have a black screen so I cannot test anything else. (on latest dev)

**Comment by ZeroCoolGOS** [comment]
@Bboy486 

Glad to see the save is working for you now, but I don't understand why your not seeing anything when it comes to running.

Its been a while since you have been up and running.  Sorry for that.

Can you provide me with a latest copy of your config (sanitized of your token)?  I will see if there is anything wrong inside the config file for it to be causing you an issue.

Also how many log files have been generated?

**Comment by Bboy486** [comment]
Custom Logs: 12
Items in cache/logs 




[configsanitized.txt](https://github.com/MattsShack/Plex-Movie-Poster-Display/files/6209412/configsanitized.txt)

**Comment by ZeroCoolGOS** [comment]
Thank you.

Is this the latest version that comes out of your PMPD after you have made adjustments such as just testing the save?

**Comment by Bboy486** [comment]
yes it is

**Comment by ZeroCoolGOS** [comment]
@Bboy486 I will try and re-review this issue over the weekend and get back to you. 

There is most likely a new build that you may want to take.
